openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/method/FilterGetMethod.scala

```diff
@@ -70,21 +70,18 @@ class FilterGetMethod @Inject()(val metricFactory: MetricFactory,
       case errors: JsError => Left(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
     }
 
-  private def parseRuleFromJavaToScala(rule: JavaRule): Rule =
-    Rule(Name(rule.getName),
-      Condition(Field(rule.getCondition.getField.asString()), Comparator(rule.getCondition.getComparator.asString()), rule.getCondition.getValue),
-      Action(AppendIn(parseMailboxIds(rule.getAction.getAppendInMailboxes.getMailboxIds))))
-
   private def parseMailboxIds(mailboxIds: ImmutableList[String]) : List[MailboxId] = List(mailboxIdFactory.fromString(mailboxIds.get(0)))
 
   private def retrieveFilters(username: Username) : SMono[Filter] =
     SFlux.fromPublisher(filteringManagement.listRulesForUser(username))
-      .map(parseRuleFromJavaToScala)
+      .map(rule => Rule(Name(rule.getName),
+        Condition(Field(rule.getCondition.getField.asString()), Comparator(rule.getCondition.getComparator.asString()), rule.getCondition.getValue),
+        Action(AppendIn(parseMailboxIds(rule.getAction.getAppendInMailboxes.getMailboxIds)))))
```

Comment:
What i meant is move the parse method in Rule class so you can do something like: .map(Rule.fromjava(...)) :)

Reply:
Actually if I move parseRuleFromJavaToScala method (to a object Rule to use static method like above, right?), i will need move parseMailboxsIds method also which use mailboxIdFactory(need to be inject). It's quite complicated

Reply:
Please have a look again to see if the code is OK

Reply:
i think fromJava() would be shorter and also more consistent with our current naming strategy.